src/daft-core/src/array/ops/list.rs

```diff
@@ -112,6 +112,67 @@ impl ListArray {
             Box::new(arrow2::array::Utf8Array::from_iter(result)),
         )))
     }
+
+    fn get_children_helper(
+        &self,
+        idx_iter: &mut impl Iterator<Item = Option<i64>>,
+        default: &Series,
+    ) -> DaftResult<Series> {
+        let default = default.cast(self.child_data_type())?;
+
+        let mut growable = make_growable(
+            self.name(),
+            self.child_data_type(),
+            vec![&self.flat_child, &default],
+            true,
+            self.len(),
+        );
+
+        // set all default indices to 0 if there is only one
+        let multiple_defaults = match default.len() {
```

Comment:
I think we can probably make an assertion that `default` has length 1 and only allow for that case from the user-API.

Should be pretty rare that a user wants to have a different default per-element!

Reply:
In Python-land you can wrap the default into a `LiteralExpression` like so:

```
def get(idx: int, default: Any):
    default_expr = lit(default)
    ...
```

Then when it gets evaluated in Rust it should evaluate the a Series with length 1 I believe (we should double-check!)

Reply:
Thanks! This method does seem to work, although it doesn't seem to emit any errors when you pass in an Expression, it just manages it as a generic PyObject. 

I think this is probably better handled on the Rust side, but is there a way to check of an expression's type is castable to another's?